frontend/src/components/Teacher/TeacherList.vue

```diff
@@ -0,0 +1,132 @@
+<script setup>
+import { computed, ref } from 'vue';
+import DeleteIcon from '@/assets/img/delete_icon.svg';
+import teacherService from '@/api/teachers';
+
+const searchQuery = ref('');
+const props = defineProps({
+  teachers: {
+    type: Array,
+    default: () => [],
+  },
+});
+const teacherFields = {
+  teacherCode: 'Code',
+  firstName: 'First name',
+  lastName: 'Last name',
+  createdAt: 'Created At',
+};
+
+const emit = defineEmits(['reload']);
+const filteredTeachers = computed(() => {
+  const filtered = props.teachers.filter(teacher =>
+    teacher.firstName.toLowerCase().includes(searchQuery.value.toLowerCase()),
+  );
```

Comment:
Lint: if prettier is setup up to do it this way then ignore this but if not then I'll mention it might be confusing to read this function at first because it is multi-line without brackets. That usually means that nothing is returned but that is not true here. So maybe make it into a single line or multi-line with brackets.
Nitpick: `searchQuery` is changed to lower case in each iteration. There is no need for that. It could be done outside of the `filter` callback so that it is done only once.

Reply:
Prettier is set up to do that, but I will rewrite it into multi-line with brackets 👍 Considering searchQuery, I noticed that after last PR but wanted to change that in last refactor PR.